gothic/gothic.go

```diff
@@ -99,6 +99,10 @@ var SetState = func(req *http.Request) string {
 // This is used to prevent CSRF attacks, see
 // http://tools.ietf.org/html/rfc6749#section-10.12
 var GetState = func(req *http.Request) string {
+	params := req.URL.Query()
+	if params.Encode() == "" && req.Method == "POST" {
```

Comment:
Consider using `http.MethodPost` here instead of hardcoding "POST".

Reply:
Yes. See https://github.com/markbates/goth/pull/356/commits/1ace300f3fc29e2458b7d7ec3ead4d33efa86968. Thank you.